helpers/checkpoint_logger/__init__.py

```diff
@@ -272,30 +271,28 @@ class MyEnum(str, Enum):
     aren't instrumented.
     """
 
-    def log_counters(obj: T) -> None:
-        metrics.incr(f"{klass.__name__}.events.{obj.name}")
-        PROMETHEUS_HANDLER.log_checkpoints(flow=klass.__name__, checkpoint=obj.name)
+    def log_counters(obj: T, context: CheckpointContext = None) -> None:
```

Comment:
```suggestion
    def log_counters(obj: T, context: CheckpointContext | None = None) -> None:
```
the same pattern repeats a bunch of times below. If the default is `None`, then `None` has to appear in the type as well.